tests/beacon/conftest.py

```diff
@@ -553,22 +621,42 @@ def genesis_state(sample_beacon_state_params,
 
 
 @pytest.fixture
-def genesis_validators(init_validator_keys,
+def initial_validators(init_validator_pubkeys,
                        init_randao,
-                       max_deposit):
+                       max_deposit,
+                       far_future_slot):
+    """
+    Inactive
+    """
     return tuple(
-        ValidatorRecord(
-            pubkey=pub,
+        mock_validator_record(
+            pubkey=pubkey,
+            far_future_slot=far_future_slot,
             withdrawal_credentials=ZERO_HASH32,
             randao_commitment=init_randao,
-            randao_layers=0,
-            status=ValidatorStatusCode.ACTIVE,
-            latest_status_change_slot=0,
-            exit_count=0,
-            custody_commitment=ZERO_HASH32,
-            latest_custody_reseed_slot=0,
-            penultimate_custody_reseed_slot=0,
-        ) for pub in init_validator_keys
+            status_flags=0,
+            is_active=False,
+        )
+        for pubkey in init_validator_pubkeys
+    )
+
+
+@pytest.fixture
+def activated_genesis_validators(initial_validators, genesis_slot):
+    """
+    Active
+    """
+    validators = tuple()
+    for validator in initial_validators:
+        validators += (validator.copy(activation_slot=genesis_slot),)
```

Comment:
*nitpick

This function *appears* to be well suited for using `eth_utils.to_tuple` and converting this to a `yield` pattern